src/app/services/contacts/contacts.ts

```diff
@@ -53,27 +53,33 @@ export class ContactsProvider {
 	}
 
 	public editContact(address: string, name: string): Observable<any> {
-		const contact = this.getContactByAddress(address);
-		if (!contact) {
+		if (!name) {
 			return throwError({
-				key: "CONTACTS_PAGE.CONTACT_NOT_EXISTS_ADDRESS",
-				parameters: { address },
+				key: "CONTACTS_PAGE.CONTACT_NAME_EMPTY",
 			} as TranslatableObject);
 		}
 
-		if (!name) {
+		if (!address) {
 			return throwError({
-				key: "CONTACTS_PAGE.CONTACT_NAME_EMPTY",
+				key: "CONTACTS_PAGE.CONTACT_ADDRESS_EMPTY",
 			} as TranslatableObject);
 		}
 
-		const existingContact = this.getContactByName(name);
-		if (existingContact && existingContact.address !== address) {
+		const contactByName = this.getContactByName(name);
+		if (contactByName && contactByName.address !== address) {
```

Comment:
You can use the [optional chaining](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining):

```suggestion
		if (contactByName?.address !== address) {
```